internal/ent/schema/apitoken.go

```diff
@@ -63,6 +63,20 @@ func (APIToken) Fields() []ent.Field {
 				token := keygen.PrefixedSecret("tola") // api token prefix
 				return token
 			}),
+		field.String("token_public_id").
+			Unique().
+			Immutable().
+			Optional().
+			Nillable().
```

Comment:
I dont think you need `Nillable()` here as this value shouldn't be cleared and should be not empty
```suggestion
			Optional().
			NotEmpty().
```